### PR TITLE
Add lazy examples.

### PR DESCRIPTION
**Pull request by MTCam**
This change set adds some lazy examples in such a way that they are automatically run by the `run_examples.sh` script used by CI.  Once we have these working, we can easily just modify `run_examples.sh` to run selected examples lazily if we want to do away with the links.

To exercise this:
> emirge/install.sh --branch=add-lazy-example
> source emirge/config/activate_env.sh
> emirge/mirgecom/examples/run_examples.sh emirge/mirgecom/examples

Currently, pulse, vortex, and wave are run in both eager and lazy modes.

cc: @kaushikcfd @matthiasdiener

### CODE REVIEW

**Comment by MTCam** [comment]
Good news everyone.  It looks like if I set up the faster tests to use lazy eval, and remove the super expensive lump test, that the increase in CI time is pretty small (~50%?).

**Comment by MTCam** [comment]
@matthiasdiener @kaushikcfd This is ready to merge whenever you get a chance to look it over.  I will take out the loopy branch merging.

**Review by matthiasdiener — APPROVED** [review]

**Comment by MTCam** [comment]
Thanks! Just need to squish the loopy branch.

File: requirements.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ psutil
 # The following packages will be git cloned by emirge:
 --editable git+https://github.com/inducer/pymbolic.git#egg=pymbolic
 #--editable git+https://github.com/inducer/pyopencl.git#egg=pyopencl
---editable git+https://github.com/inducer/loopy.git#egg=loopy
+--editable git+https://github.com/inducer/loopy.git@kernel_schedule_is_a_tree_pls_no_linear_representations#egg=loopy
```
Comment (kaushikcfd) [inline requirements.txt:13]:
```suggestion
--editable git+https://github.com/inducer/loopy.git#egg=loopy
```

**Review by kaushikcfd — APPROVED** [review]
Except the loopy branch, this LGTM!

**Pull request merged**